src/couch/src/couch_file.erl

```diff
@@ -272,14 +286,18 @@ deleted_filename(Original) ->
             ++ filename:extension(Original), [Y, Mon, D, H, Min, S])),
     filename:rootname(Original) ++ Suffix.
 
-nuke_dir(RootDelDir, Dir) ->
+nuke_dir(RootDir, Dir) ->
+    nuke_dir(RootDir, Dir, []).
+nuke_dir(RootDelDir, Dir, Options) ->
     EnableRecovery = config:get_boolean("couchdb",
         "enable_database_recovery", false),
+    Context = couch_util:get_value(context, Options, compaction),
     case EnableRecovery of
-        true ->
-            rename_file(Dir);
-        false ->
-            delete_dir(RootDelDir, Dir)
+        true when Context == delete ->
+            DbName = couch_util:get_value(db_name, Options),
+            rename_dir(RootDelDir, Dir, DbName);
+        true -> rename_file(Dir);
```

Comment:
Ah, gotchya. For some reason I was thinking that was somewhere else but I forgot what module i was reading.